util/tracing/visualization/fedsd.py

```diff
@@ -334,9 +334,12 @@ def load_and_process_csv_file(csv_file) :
     # which boils up to having 'RTI' in the 'event' column
     df = df[df['event'].str.contains('Sending|Receiving|Scheduler advancing time ends') == True]
 
-    # Fix the parameters of the event 'Scheduler advancing time ends'
-    # We rely on the fact that the first row of the csv file cannot be the end of advancing time
-    id = df.iloc[-1]['self_id']
+    # Determine the "self id" in the trace file based on the first 'Receiving' or 'Sending' message (or use -1, the id of the RTI, if there is none).
```

Comment:
```suggestion
    # Determine the "self id" in the trace file based on the first 'Receiving' message (or use -1, the id of the RTI, if there is none).
```
Looks like we're not checking for 'Sending' messages?

Reply:
Looks to me like the code checks for both.

Reply:
We do!
The `if` statement in the `for` loop says (Line 340):
```
if ('Sending' in row['event'] or 'Receiving' in row['event']) :
``` 
that is, if `row['event']` has `Sending` *OR* `Receiving` in it.

Reply:
Oh, missed that. Looks good!